Sources/MapboxCoreNavigation/PredictiveCacheManager.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import MapboxNavigationNative
+
+/**
+ `PredictiveCacheManager` is responsible for creating and retaining `Predictive Caching` related components.
+ 
+ Typical usage suggests initializing an instance of `PredictiveCacheManager` and retaining it as long as caching is required.
+ */
+public class PredictiveCacheManager {
+    public typealias MapOptions = (tileStore: TileStore, styleSourcePaths: [String])
+    
+    private(set) var controllers: [PredictiveCacheController] = []
+    
+    /**
+     Default initializer
+     
+     - parameter predictiveCacheOptions: `PredictiveCacheOptions` which configures various caching parameters like radiuses of current and destination locations.
+     - parameter mapOptions: A `MapOptions` which contains info about `MapView` tiles like its location and tilesets to be cached. If set to `nil` - predictive caching won't be enbled for map tiles.
```

Comment:
In general, the type of a parameter doesn’t need to be mentioned explicitly, because jazzy and Quick Help will already display and link the type automatically.

```suggestion
     Initializes a predictive cache.
     
     - parameter predictiveCacheOptions: A configuration specifying various caching parameters, such as the radii of current and destination locations.
     - parameter mapOptions: Information about `MapView` tiles such as the location and tilesets to cache. If this argument is set to `nil`, predictive caching is disabled for map tiles.
```